packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,632 @@
+  /// Background color of the container while it is open.
+  ///
+  /// When the container is opened, it will transition from [closedColor] to
+  /// this color via [Colors.white]. When the container is closed, it will
+  /// transition from this color back to [closedColor] via [Colors.white].
+  ///
+  /// Defaults to [Colors.white].
+  ///
+  /// See also:
+  ///
+  ///  * [Material.color], which is used to implement this property.
+  final Color openColor;
+
+  /// Elevation of the container while it is closed.
+  ///
+  /// When the container is opened, it will transition from this elevation to
+  /// [openElevation]. When the container is closed, it will transition back
+  /// from [openElevation] to this elevation.
+  ///
+  /// Defaults to 1.0.
+  ///
+  /// See also:
+  ///
+  ///  * [Material.elevation], which is used to implement this property.
+  final double closedElevation;
+
+  /// Elevation of the container while it is open.
+  ///
+  /// When the container is opened, it will transition to this elevation from
+  /// [closedElevation]. When the container is closed, it will transition back
+  /// from this elevation to [closedElevation].
+  ///
+  /// Defaults to 4.0.
+  ///
+  /// See also:
+  ///
+  ///  * [Material.elevation], which is used to implement this property.
+  final double openElevation;
+
+  /// Shape of the container while it is closed.
+  ///
+  /// When the container is opened it will transition from this shape to a
+  /// rectangular that fills the surrounding [Navigator]. When the container
+  /// is closed, it will transition back to this shape.
+  ///
+  /// Defaults to a [RoundedRectangleBorder] with a [Radius.circular] of 4.0.
+  ///
+  /// See also:
+  ///
+  ///  * [Material.shape], which is used to implement this property.
+  final ShapeBorder closedShape;
+
+  /// Called to obtain the child for the container in the closed state.
+  ///
+  /// The [Widget] returned by this builder is faded out when the container
+  /// opens and at the same time the [Widget] returned by [openBuilder] is faded
+  /// in while the container grows to fill the surrounding [Navigator].
+  ///
+  /// The `action` callback provided to the builder can be called to open the
+  /// container.
+  final OpenContainerChildBuilder closedBuilder;
+
+  /// Called to obtain the child for the container in the open state.
+  ///
+  /// The [Widget] returned by this builder is faded in when the container
+  /// opens and at the same time the [Widget] returned by [closedBuilder] is
+  /// faded out while the container grows to fill the surrounding [Navigator].
+  ///
+  /// The `action` callback provided to the builder can be called to close the
+  /// container.
+  final OpenContainerChildBuilder openBuilder;
+
+  /// Whether the entire closed container can be tapped to open it.
+  ///
+  /// Defaults to true.
+  ///
+  /// When this is set to false the container can only be opened via the
+  /// `action` callback provided to [closedBuilder].
```

Comment:
to openBuilder? This is a bit confusing. How would calling a method that I also passed as an action callback, open or close the container?

Reply:
I rephrased this a little.

What I want to say is: the closedBuilder that you provide is passed an action callback as argument. When you call that callback it will open the container. (Your closedBuilder will probably have a GestureDetector in the widget hierarchy it returns and that GestureDetector has its onTap set to that action callback.)